File: src/js/pe_0008.js
// FUNCTIONS


function prod(arr) {
    return arr.reduce((a, b) => a * b, 1);
}


// SCRIPT


const fh = require('fs')
const text = fh.readFileSync("data/pe_0008.txt", "utf8")
let arr = [0, 0, 0, 0];

let i = 0;
let hold = 0;

for (let j = 0; j < text.length; j++) {
    let c = text[j];
    if ('0' <= c && c <= '9') {
        arr[i % 4] = parseInt(c);
        let p = prod(arr);
        if (hold < p) {
            hold = p;
        }
        i++;
    }
}

console.log(hold);

File: src/py/pe_0008.py
import functools
import operator


# FUNCTIONS


def prod(itr):
    return functools.reduce(operator.mul, itr, 1)


# SCRIPT


if __name__ == "__main__":
    fpath = "data/pe_0008.txt"
    fh = open(fpath, encoding="utf8")

    i = 0
    hold = 0
    lst = [0, 0, 0, 0]
    for line in map(str.strip, fh):
        for x in line:
            n = int(x)

            lst[i % 4] = n
            p = prod(lst)
            if hold < p:
                hold = p

            i += 1

    fh.close()
    print(hold)
